From 28e71f65c3183953072b51b6a8fd4d6d0a9d5bbd Mon Sep 17 00:00:00 2001
From: Dan Prince <dprince@redhat.com>
Date: Wed, 3 Sep 2025 14:42:52 -0400
Subject: [PATCH 1/5] Intitial OLM bundle for the cluster-operator

This patch also updates PROJECT to drop the group:
name as this shouldn't be required for this operator
which doesn't currently use group names (and causes
the bundle to fail to generate)

Remove Icon from bundle

Add required infrastructure annotations

Signed-off-by: Dan Prince <dprince@redhat.com>
---
 PROJECT                                       |  5 +-
 ...luster-operator.clusterserviceversion.yaml | 51 +++++++++++++++++++
 config/manifests/kustomization.yaml           | 10 ++++
 hack/build-olm-bundle.sh                      | 15 ++++++
 properties.yaml                               |  4 ++
 5 files changed, 83 insertions(+), 2 deletions(-)
 create mode 100644 config/manifests/bases/rabbitmq-cluster-operator.clusterserviceversion.yaml
 create mode 100644 config/manifests/kustomization.yaml
 create mode 100644 hack/build-olm-bundle.sh
 create mode 100644 properties.yaml

diff --git a/PROJECT b/PROJECT
index c695f73b..1da07372 100644
--- a/PROJECT
+++ b/PROJECT
@@ -1,7 +1,9 @@
 domain: rabbitmq.com
 layout:
 - go.kubebuilder.io/v3
-projectName: cluster-operator
+plugins:
+  manifests.sdk.operatorframework.io/v2: {}
+projectName: rabbitmq-cluster-operator
 repo: github.com/rabbitmq/cluster-operator
 resources:
 - api:
@@ -9,7 +11,6 @@ resources:
     namespaced: true
   controller: true
   domain: rabbitmq.com
-  group: rabbitmq.com
   kind: RabbitmqCluster
   path: github.com/rabbitmq/cluster-operator/api/v1beta1
   version: v1beta1
diff --git a/config/manifests/bases/rabbitmq-cluster-operator.clusterserviceversion.yaml b/config/manifests/bases/rabbitmq-cluster-operator.clusterserviceversion.yaml
new file mode 100644
index 00000000..a885cd5b
--- /dev/null
+++ b/config/manifests/bases/rabbitmq-cluster-operator.clusterserviceversion.yaml
@@ -0,0 +1,51 @@
+apiVersion: operators.coreos.com/v1alpha1
+kind: ClusterServiceVersion
+metadata:
+  annotations:
+    alm-examples: '[]'
+    capabilities: Basic Install
+    features.operators.openshift.io/disconnected: "true"
+    features.operators.openshift.io/fips-compliant: "true"
+    features.operators.openshift.io/proxy-aware: "false"
+    features.operators.openshift.io/tls-profiles: "false"
+    features.operators.openshift.io/token-auth-aws: "false"
+    features.operators.openshift.io/token-auth-azure: "false"
+    features.operators.openshift.io/token-auth-gcp: "false"
+    operators.operatorframework.io/operator-type: non-standalone
+  name: rabbitmq-cluster-operator.v0.0.1
+  namespace: placeholder
+spec:
+  apiservicedefinitions: {}
+  customresourcedefinitions:
+    owned:
+    - description: RabbitmqCluster is the Schema for the RabbitmqCluster API. Each
+        instance of this object corresponds to a single RabbitMQ cluster.
+      displayName: Rabbitmq Cluster
+      kind: RabbitmqCluster
+      name: rabbitmqclusters.rabbitmq.com
+      version: v1beta1
+  description: RabbitMQ Cluster Operator
+  displayName: RabbitMQ Cluster Operator
+  install:
+    spec:
+      deployments: null
+    strategy: ""
+  installModes:
+  - supported: true
+    type: OwnNamespace
+  - supported: true
+    type: SingleNamespace
+  - supported: false
+    type: MultiNamespace
+  - supported: true
+    type: AllNamespaces
+  keywords:
+  - rabbitmq
+  links:
+  - name: Cluster Operator
+    url: https://github.com/rabbitmq/cluster-operator
+  maturity: alpha
+  provider:
+    name: RabbitMQ
+    url: https://rabbitmq.com
+  version: 0.0.1
diff --git a/config/manifests/kustomization.yaml b/config/manifests/kustomization.yaml
new file mode 100644
index 00000000..69044879
--- /dev/null
+++ b/config/manifests/kustomization.yaml
@@ -0,0 +1,10 @@
+# These resources constitute the fully configured set of manifests
+# used to generate the 'manifests/' directory in an OLM bundle.
+# Required to use the hack/make-olm-bundle.sh script via
+# operator-sdk.
+resources:
+- bases/rabbitmq-cluster-operator.clusterserviceversion.yaml
+- ../default/base
+- ../samples/base
+- ../crd
+- ../rbac
diff --git a/hack/build-olm-bundle.sh b/hack/build-olm-bundle.sh
new file mode 100644
index 00000000..306dcf67
--- /dev/null
+++ b/hack/build-olm-bundle.sh
@@ -0,0 +1,15 @@
+# Script to build an OLM bundle (Operator Lifecycle Manager)
+# NOTE: requires operator-sdk
+
+set -x
+rm -Rf bundle bundle.Dockerfile
+make manifests
+operator-sdk generate kustomize manifests -q
+#cd config/manager && kustomize edit set image controller=rabbitmq-cluster-operator:latest
+kustomize build config/manifests | operator-sdk generate bundle --overwrite --verbose --use-image-digests
+#FIXME: look into why scorecard isn't being deactivated
+sed -e '/.*scorecard/d' -i bundle.Dockerfile
+operator-sdk bundle validate ./bundle
+
+echo "Listing..."
+ls bundle/manifests
diff --git a/properties.yaml b/properties.yaml
new file mode 100644
index 00000000..82c2ebb4
--- /dev/null
+++ b/properties.yaml
@@ -0,0 +1,4 @@
+properties:
+  - property:
+      type: product
+      value: openstack
-- 
2.51.0

